### DISCUSSION TITLE
Does serverpod support string ids ?

### DISCUSSION THREAD

**Discussion by h-er0** [serverpod/serverpod#2708] [Q&A] [answered]
I need to store external data using non-integer identifiers. Will serverpod support String ids in the future ?

**Comment by vlidholt** [comment, marked as answer]
The way to do this is to create a field (column) with a unique index. You will still have the numerical `id` field, as Serverpod needs it for some of its operations.

**Comment by tananga** [comment]
@h-er0 Base on this PR https://github.com/serverpod/serverpod/pull/3257 you can think that yes but I know from Marcelo that right now only int and UUID works.
